sources/packages/api/src/route-controllers/institution/institution.controller.ts

```diff
@@ -259,4 +264,50 @@ export class InstitutionController extends BaseController {
       body.isActive,
     );
   }
+
+  @Get("/my-details")
+  async getMyDetails(
+    @UserToken() token: IInstitutionUserToken,
+  ): Promise<InstitutionUserAndAuthDetails> {
+    // Get logged in user and location details with auth
+    const institutionUser =
+      await this.institutionService.getInstitutionUserByUserName(
+        token.userName,
+      );
+    const user = {
+      user: {
+        firstName: institutionUser.user?.firstName,
+        lastName: institutionUser.user?.lastName,
+        userName: token.userName ?? institutionUser.user?.userName,
```

Comment:
I believe that we should not expose this in the API. Is there any use of this outside the API?
The API will always have this information on the koken and the Vue application or other possible API client should not send it to the API.

Reply:
client-side token has logged in user's username only. In header, we need logged in user information. 
![image](https://user-images.githubusercontent.com/77353155/121941116-dfa76500-cd1c-11eb-975f-c988914b8b58.png)